### PR TITLE
Add examples to http_wwwauth.xml

### PR DESCRIPTION
**Pull request by tsellers-r7**
## Description
I'm submitting this PR since I happened to have the data handy.  This PR only updates `xml/http_wwwauth.xml` and makes the following changes:

- Adds some missing examples
- Adds 1 fingerprint for SpeedTouch and reworks the WRT54G fingerprint to be more global
- Replaces the wildcard `.` with `&quot;` where appropriate in the regex
- Adds `hw.*` params in select cases where I was already touching a fingerprint.
- Moves `Fujitsu Siemens Primergy with BMC RemoteView on an iRMC card` lower in the file behind more frequently seen matches.

## Motivation and Context
Improved ability test.

## How Has This Been Tested?
`rspec`

## Types of changes

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have updated the documentation accordingly (or changes are not required).
- [x] I have added tests to cover my changes (or new tests are not required).
- [x] All new and existing tests passed.

### CODE REVIEW

File: xml/http_wwwauth.xml [hunk 1]
```diff
@@ -38,9 +33,13 @@
     <param pos="0" name="os.product" value="IOS"/>
     <param pos="0" name="os.version" value="12"/>
     <param pos="0" name="os.cpe23" value="cpe:/o:cisco:ios:12"/>
+    <param pos="0" name="hw.vendor" value="Cisco"/>
+    <param pos="0" name="hw.device" value="Router"/>
   </fingerprint>
   <fingerprint pattern="^(?:Basic|Digest) realm=.level[ _]\d\d?[ _]access.$">
```
Comment (jhart-r7) [inline xml/http_wwwauth.xml]: Why no quote change here?

File: xml/http_wwwauth.xml [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
-    <param pos="0" name="service.vendor" value="Fujitsu Siemens"/>
-    <param pos="0" name="service.product" value="RemoteView"/>
-    <param pos="0" name="service.family" value="RemoteView"/>
-    <param pos="0" name="hw.vendor" value="Fujitsu Siemens"/>
-    <param pos="0" name="hw.family" value="Primergy"/>
-    <param pos="0" name="hw.product" value="Primergy"/>
-    <param pos="1" name="host.name"/>
-  </fingerprint>
-  <fingerprint pattern="^(?:Basic|Digest) realm=.access.$">
+  <fingerprint pattern="^(?:Basic|Digest) realm=&quot;access&quot;$">
```
Comment (jhart-r7) [inline xml/http_wwwauth.xml]: Hmm, will it always be double quotes?  Perhaps that it is why historically `.` has been used to may be match single or double quotes?

Comment (tsellers-r7) [reply xml/http_wwwauth.xml]: I should have worded that differently. I had replaced them opportunistically.  I've gone back now and replaced them all. Thanks for the feedback.

File: xml/http_wwwauth.xml (hunk 1 above)
Comment (tsellers-r7) [reply xml/http_wwwauth.xml]: I should have worded that differently. I had replaced them opportunistically. I've gone back now and replaced them all. Thanks for the feedback.

File: xml/http_wwwauth.xml (hunk 2 above)
Comment (tsellers-r7) [reply xml/http_wwwauth.xml]: Good point. We actually have examples of this in our data set but none that are currently covered by the existing database.

Comment (tsellers-r7) [reply xml/http_wwwauth.xml]: And I immediately see that I'm wrong.

Comment (tsellers-r7) [reply xml/http_wwwauth.xml]: Should be sorted out now

File: xml/http_wwwauth.xml [hunk 3]
```diff
@@ -167,18 +170,35 @@
...
     <param pos="0" name="os.device" value="Broadband router"/>
     <param pos="0" name="os.family" value="SpeedTouch"/>
     <param pos="0" name="os.product" value="SpeedTouch"/>
+    <param pos="0" name="hw.vendor" value="Thomson"/>
+    <param pos="0" name="hw.family" value="SpeedTouch"/>
+    <param pos="0" name="hw.device" value="Broadband router"/>
     <param pos="1" name="host.mac"/>
   </fingerprint>
-  <fingerprint pattern="^(?:Basic|Digest) realm=.SpeedTouch., nonce=.[0-9A-Z]+:([0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}):\d+:\d+., qop=.auth.$">
+  <fingerprint pattern="^(?:Basic|Digest) realm=&quot;SpeedTouch., nonce=.[0-9A-Z]+:([0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}-[0-9A-F]{2}):\d+:\d+., qop=.auth&quot;$">
```
Comment (jhart-r7) [inline xml/http_wwwauth.xml]: This one and the one above it are identical with the exception of the `qop` part which isn't reflected in the resulting fingerprint anyway.  Combine into one.

Comment (tsellers-r7) [reply xml/http_wwwauth.xml]: Yup, looks like one was derived from Digest and the other basic.

**Pull request merged**